### ISSUE TITLE
Training not working, issue with mismatch tensor.

### ISSUE THREAD

**Issue by newstown24** [myshell-ai/MeloTTS#292]
Using noise scaled MAS for VITS2
Using duration discriminator for VITS2
/home/rohan/Desktop/Projects/MeloTTS/env/lib/python3.10/site-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
(torch.Size([10, 192]), torch.Size([8, 192]))
(torch.Size([256, 256]), torch.Size([1, 256]))
list index out of range
/home/rohan/Desktop/Projects/MeloTTS/melo/train_new.py:252: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = GradScaler(enabled=hps.train.fp16_run)
0it [00:00, ?it/s]/home/rohan/Desktop/Projects/MeloTTS/env/lib/python3.10/site-packages/torch/utils/data/dataloader.py:626: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 12, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
0it [00:00, ?it/s]
/home/rohan/Desktop/Projects/MeloTTS/env/lib/python3.10/site-packages/torch/optim/lr_scheduler.py:182: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn(
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]
0it [00:00, ?it/s]

**Comment by KiON-GiON** [comment]
This is a common problem with MeloTTS. Are you finetuning the base checkpoint or training a voice from scratch? If the parameters don't match, they'll be reinitialized, meaning that you will be training it from scratch. Make sure to modify the config values to the same as the base model if you're finetuning.
Important values to modify are "num_tones", "num_languages" and "n_speakers". At least that's what I did when finetuning a checkpoint. My finetune has 3 speakers but I set "n_speakers" to 256 like the base checkpoint.

**Comment by newstown24** [comment]
Hi @KiON-GiON (torch.Size([10, 192]), torch.Size([8, 192])) this error is still persistent after modifying the base config.

**Comment by KiON-GiON** [comment]
@newstown24 Are you modifying the base config or the generated config after running the preprocessing step? You have to modify the generated config. In my case, I had to match the order of the keys. If "num_languages" and "num_tones" are in another section of the JSON (for some reason), move them to the end of the symbol list. For example:
```
    "symbols": [
...
        "?",
        "\u2026",
        ",",
        ".",
        "'",
        "-",
        "\u00bf",
        "\u00a1",
        "SP",
        "UNK"
    ],
    "num_tones": 16,
    "num_languages": 10
}
```
I had to do this because the original config of the base model I used had this order too.
